### ISSUE TITLE
Focus leaks between radio groups

### ISSUE THREAD

**Issue by Jinnie** [vmware/clarity#3644]
## Describe the bug

If we have adjacent radio groups two undesired behaviors can be observed:
- Moving focus with Tab skips the 2nd group.
- Focus can be moved with keys from first to second group.

## How to reproduce

https://stackblitz.com/edit/clarity-radios-study?file=src/app/app.component.html

1.  Go to the provided [stackblitz](https://stackblitz.com/edit/clarity-radios-study?file=src/app/app.component.html)
2.  Follow the instructions in orange.

## Expected behavior

Focus should be managed per group and each group should be separately reachable by Tabbing.
If end of one group is reached when using arrow keys then focus should move to the first element of the same group.

## Versions

**App**

* Clarity: 2.0.2

**Device:**

* Type: MacBook
* Browser chrome

**Comment by Jinnie** [comment]
Just noticed that not only focus but selection also is "shared" between the 2 radio groups, though this may be caused by the lack of link to a model in the demo.

**Comment by gnomeontherun** [comment]
This is because the radios have the same `name="options"`, it should be unique between different groups. That would fix this issue. Closing as expected, since HTML will treat them as a group when they have the same name.
